File: 347. Top K Frequent Elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        res = {}
        returnList = []

        for x in nums:
            if x not in res:
                nums.count(x)
                res[x] = nums.count(x)
        for x in range(k):
            g = max(res, key=res.get)
            returnList.append(g)
            del res[g]

        return returnList

        

File: 347. Top K Frequent Elements.cpp

#include <iostream>
#include <string>
#include <unordered_set>
#include <vector>
#include <map>
class Solution {
public:
    std::vector<int> topKFrequent(std::vector<int>& nums, int k) {
        std::map<int, int> counters;  
        std::vector<int> return_vector;
        for(auto i: nums)
        {
            ++counters[i];
        }
       std::vector<std::vector<int>> bucket(nums.size()+1);
        for(auto& [key,value]: counters){
            bucket[value].push_back(key);
        }
        for(int i= bucket.size()-1; i>=0; i--){
            for(auto n: bucket[i]){
                if(k>0){
                    return_vector.push_back(n);
                    k--;
                }
            }
        }
        return return_vector;
    }
   


};